### PR TITLE
Predator Rework

### PR DESCRIPTION
**Pull request by EnvyDragon**
Generator PR: https://github.com/Ziktofel/Archipelago/pull/262

Changes:
- Cloak (add energy bar, same functionality otherwise) is now Phase Cloak (no energy cost, 5 second duration, 15 second timer, ignores collision while active)
- Predator's Fury is now Vespene Synthesis (gain 1 vespene every time Retribution field hits something)
- Charge is now Concussive Charge (next attack after charging applies an AoE slow)

New upgrade:
- Adaptive Defenses: Predator gains 50 maximum shields. 15% of Retribution Field damage recovers those shields. Ranged and Splash damage is reduced by 50% while shields are active (so they can, for example, survive 1 Siege Tank hit or 2 Baneling attacks and still get out without taking health damage; the shield leech also means enemy ranged units near targets of opportunity aren't dealbreakers to the same degree. Mistakes are much, MUCH less punishing with this upgrade)

Other changes:
- Vehicle Weapon upgrades also add 2 per level to Retribution Field damage. Retribution Field tooltip has been modified to check the ability's real damage.
- Vehicle Armor upgrades also add 1 per level to Predator Shield Armor.

### CODE REVIEW

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UnitData.xml [hunk 1]
```diff
@@ -3184,7 +3185,10 @@
         <AbilArray Link="stop"/>
         <AbilArray Link="AP_PredatorCloak"/>
         <AbilArray Link="AP_ChargePredator"/>
-        <AbilArray Link="AP_PredatorFury"/>
+<!--        <AbilArray Link="AP_PredatorFury"/>-->
```
Comment (MatthewMarinets) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UnitData.xml:3188]: Can remove commented line.

**Review by MatthewMarinets — APPROVED** [review]
Skimmed, didn't look too far into the validator mess of Fury, but looks good from what I saw. I like the detail of giving the shield armour display the golden shield icon, though it means the icon doesn't change with upgrades. If you want to change that, some lower-level icon possibilities:
* btn-ability-protoss-invulnerabilityshield.dds
* btn-ability-protoss-hardenedshields-color.dds
* btn-ability-protoss-shieldcapacitor.dds
* Current: btn-upgrade-stetmann-zerglinghardenedshield.dds
* btn-upgrade-stetmann-banelingmanashieldefficiency.dds

btn-ability-protoss-invulnerabilityshield
btn-ability-protoss-hardenedshields-color
btn-ability-protoss-shieldcapacitor
btn-upgrade-stetmann-zerglinghardenedshield

btn-upgrade-stetmann-banelingmanashieldefficiency


(this only brings the count to 3, though. I think it would be fine to stop changing the display after 3)

**Comment by EnvyDragon** [comment]
Reworked the rework; buffed Charge, heavily buffed Adaptive Defenses, and swapped the fury thing with Vespene Synthesis. Having done a bunch of testing, unlocking a few upgrades for them means they now feel like a unit that you can _actually_ plan a composition around in some missions, and even feel like something you'll occasionally want to build a few of for supporting purposes.

**Comment by Ziktofel** [comment]
Snímek obrazovky pořízený 2024-08-18 15-42-00
I'd add a note that shields have 2 base armor.
Also I'd harmonize the shield name with other Terran shielded units (also the upgrade icons)

TODO: Missing shield wireframe (For BCs and Medivacs it's already made)

**Comment by EnvyDragon** [comment]
> Also I'd harmonize the shield name with other Terran shielded units (also the upgrade icons)

I'll change the Predator shield name to the one used by BCs and Medivacs, that's a good point.

For the icons though, I think it's a bit strange for Terran units to use the same ones as Protoss units; would you be okay if I put the Predator shield icons on the other units instead?

(I'll apply the rest of the listed changes too: adding the 2 base armor to the tooltip, and adding a TODO for the shield wireframe)

**Comment by EnvyDragon** [comment]
> Normally Terran units that have shields in vanilla use Protoss L0 icon for them

Yeah, and I think that's kinda weird, which is why I picked a different set of upgrade icons.

Like I said, I _can_ put the Protoss icons on the Predator, but I think it would look better for Terran shields to have their own icon set.

**Comment by Ziktofel** [comment]
> > Normally Terran units that have shields in vanilla use Protoss L0 icon for them
> 
> Yeah, and I think that's kinda weird, which is why I picked a different set of upgrade icons.
> 
> Like I said, I _can_ put the Protoss icons on the Predator, but I think it would look better for Terran shields to have their own icon set.

The thing is that if you use the standard ones, it'd be consistent with Tosh/etc. Using the same icon is simple but consistent

**Comment by Ziktofel** [comment]
Also add the shield hitter effect:
Snímek obrazovky pořízený 2024-08-19 20-13-58

**Comment by EnvyDragon** [comment]
Wireframes and shield actors are added, tested, and working!

File: Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt [hunk 2]
```diff
@@ -2562,7 +2570,8 @@ Button/Tooltip/AP_PlanetaryFortressLoad=Loads nearby SCVs into the Planetary For
 Button/Tooltip/AP_PointDefenseDrone=Uses a laser to shoot down enemy missiles.  <n/>Cannot target special attacks. <n/>Times out after <d ref="Behavior,AP_PointDefenseDroneTimedLife,Duration"/> seconds.<n/><n/><c val="f078ff">Each shot consumes 10 energy.</c>
 Button/Tooltip/AP_PoisonNova=Instantly deals 20 damage to enemy ground units nearby.
 Button/Tooltip/AP_Predator=Anti-infantry specialist that deals area damage with each attack.<n/><n/><c val="#ColorAttackInfo">Can attack ground units.</c>
-Button/Tooltip/AP_PredatorFury=Jump from target to target, dealing <d ref="Effect,AP_DarkTemplarShadowFuryDamage,Amount"/> damage with each jump. Hits <d ref="Behavior,AP_PredatorShadowFuryController,PeriodCount+1"/> times.
+Button/Tooltip/AP_PredatorVespeneSynthesis=The player owning the Predator gains 1 Vespene every time Retribution Field deals damage.
+Button/Tooltip/AP_PredatorAdaptiveDefenses=Predators gain recharging shields, and take 50% reduced splash and ranged damage while the shield is active. A portion of damage dealt by Retribution Field also restores shields.<n/><n/>Shields have 2 base armor and are improved by Terran Vehicle Armor upgrades.
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt:2574]: Wording: The upgrade tooltips say `Retribution Field`, while the base trait says `Lightning Field`
Snímek obrazovky pořízený 2024-08-31 18-07-02
Snímek obrazovky pořízený 2024-08-31 18-06-57

**Review by Ziktofel — CHANGES_REQUESTED** [review]

File: Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt (hunk 2 above)
Comment (EnvyDragon) [reply Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt]: Fixed

**Pull request merged**